### PR TITLE
[GDAL] Bump to 3.7.3

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

File: G/GDAL/build_tarballs.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@
 using BinaryBuilder, Pkg
 
 name = "GDAL"
-upstream_version = v"3.7.2"
+upstream_version = v"3.7.3"
```
Comment (giordano) [inline G/GDAL/build_tarballs.jl:6]: Need to change the source as well, no?

Comment (jeremiahpslewis) [reply G/GDAL/build_tarballs.jl:6]: Definitely! Oof.

**Comment by giordano** [comment]
A linker error is unlikely to be magically fixed by changing compiler version, especially if this is linking to OpenMPI 5 which deosn't exist at all for 32-bit systems (but I can't see the log, so can't know whether that's the case)

**Comment by jeremiahpslewis** [comment]
Error: ``/opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/9.1.0/../../../../i686-linux-gnu/bin/ld: warning: libmpi.so.40, needed by /workspace/destdir/lib/libhdf5.so.310, not found (try using -rpath or -rpath-link)``

**Pull request closed without merging**